core/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodesInfoResponse.java

```diff
@@ -84,6 +86,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 }
             }
 
+            builder.array("roles", nodeInfo.getNode().getRoles().stream().map(DiscoveryNode.Role::getRoleName)
```

Comment:
can we just use a for loop here, I think we should not mix streams in too much :)